src/main/java/org/gridsuite/directory/server/DirectoryService.java

```diff
@@ -174,59 +170,37 @@ private void emitDirectoryChanged(UUID directoryUuid, String userId, String erro
-                .ifPresent(directoryElementEntity -> {
-                    boolean isPrivate = isPrivateForNotification(directoryElementEntity.getParentId(), directoryElementEntity.isPrivate());
-                    emitDirectoryChanged(
-                        directoryElementEntity.getParentId() == null ? elementUuid : directoryElementEntity.getParentId(),
-                        userId,
-                        isPrivate,
-                        directoryElementEntity.getParentId() == null,
-                        NotificationType.UPDATE_DIRECTORY
-                    );
-                });
-        });
-    }
-
-    public Mono<Void> setAccessRights(UUID elementUuid, boolean newIsPrivate, String userId) {
-        return getElementInfos(elementUuid).flatMap(elementAttributes -> {
-            if (!userId.equals(elementAttributes.getOwner())) {
-                return Mono.error(new DirectoryException(NOT_ALLOWED));
-            }
-            directoryElementRepository.updateElementAccessRights(elementUuid, newIsPrivate);
-            directoryElementRepository.findById(elementUuid).ifPresent(directoryElementEntity -> {
-                boolean isPrivate = isPrivateForNotification(directoryElementEntity.getParentId(), false);
-                // second parameter should be always false because when we pass a root folder from public -> private we should notify all clients
+    public Mono<Void> updateElement(UUID elementUuid, ElementAttributes newElementAttributes, String userId) {
+        return getElementEntity(elementUuid)
+            .filter(elementEntity -> userId.equals(elementEntity.getOwner()))
+            .switchIfEmpty(Mono.error(new DirectoryException(NOT_ALLOWED)))
+            .filter(e -> DirectoryElementEntity.isAttributesUpdatable(newElementAttributes))
+            .switchIfEmpty(Mono.error(new DirectoryException(NOT_ALLOWED)))
+            .map(e -> e.update(newElementAttributes))
+            .map(directoryElementRepository::save)
+            .doOnSuccess(elementEntity ->
                 emitDirectoryChanged(
-                    directoryElementEntity.getParentId() == null ? directoryElementEntity.getId() : directoryElementEntity.getParentId(),
+                    elementEntity.getParentId() == null ? elementUuid : elementEntity.getParentId(),
                     userId,
-                    isPrivate,
-                    directoryElementEntity.getParentId() == null,
-                    NotificationType.UPDATE_DIRECTORY);
-            });
-            return Mono.empty();
-        });
+                    // second parameter should be always false when we change the access mode of a folder because we should notify all clients
+                    isPrivateForNotification(elementEntity.getParentId(), newElementAttributes.getAccessRights() == null && elementEntity.isPrivate()),
+                    elementEntity.getParentId() == null,
+                    UPDATE_DIRECTORY
```

Comment:
See previous comment about use of NotificationType.UPDATE_DIRECTORY

Reply:
Done